game/scripts/vscripts/Sections/Chapter5/SectionRoshan.ts

```diff
@@ -50,48 +49,41 @@ function onStart(complete: () => void) {
 
     const playerHero = getOrError(getPlayerHero())
 
-    let roshan = Entities.FindAllByName(CustomNpcKeys.Roshan)[0] as CDOTA_BaseNPC
-
     // DK lvl 25 talents
     const dragonBlood25Talent = playerHero.FindAbilityByName("special_bonus_unique_dragon_knight")
     const dragonTail25Talent = playerHero.FindAbilityByName("special_bonus_unique_dragon_knight_2")
 
+    let roshan = Entities.FindAllByName(CustomNpcKeys.Roshan)[0] as CDOTA_BaseNPC
+
     graph = tg.withGoals(_ => goalTracker.getGoals(),
         tg.seq([
-            tg.immediate(() => canPlayerIssueOrders = false),
+            tg.immediate(_ => setUnitPacifist(roshan, true)),
             tg.wait(1),
-            tg.immediate(() => setUnitPacifist(roshan, true)),
             tg.fork([
+                tg.audioDialog(LocalizationKey.Script_5_Roshan_1, LocalizationKey.Script_5_Roshan_1, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
                 tg.seq([
-                    tg.audioDialog(LocalizationKey.Script_5_Roshan_1, LocalizationKey.Script_5_Roshan_1, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
                     tg.immediate(() => goalEnterRoshPit.start()),
                     tg.goToLocation(roshPitGoalPosition),
-                ]),
-                tg.seq([
-                    tg.panCameraLinear(_ => getPlayerCameraLocation(), roshPitGoalPosition, 1),
-                    tg.wait(2),
-                    tg.immediate(() => canPlayerIssueOrders = true),
-                    tg.setCameraTarget(undefined),
-                    tg.immediate(_ => centerCameraOnHero()),
-                ]),
+                    tg.immediate(() => goalEnterRoshPit.complete()),
+                ])
             ]),
-            tg.immediate(() => goalEnterRoshPit.complete()),
             tg.audioDialog(LocalizationKey.Script_5_Roshan_2, LocalizationKey.Script_5_Roshan_2, ctx => ctx[CustomNpcKeys.SunsFanMudGolem]),
-            tg.audioDialog(LocalizationKey.Script_5_Roshan_3, LocalizationKey.Script_5_Roshan_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]),
-            tg.immediate(() => {
-                // Lvl up to 25, assuming 1 xp per level
-                playerHero.AddExperience(25 - playerHero.GetLevel(), ModifyXpReason.UNSPECIFIED, true, false)
-                shared.preRoshKillItems.forEach(itemName => playerHero.AddItemByName(itemName))
-                // Apply DD rune with infinite duration
-                playerHero.AddNewModifier(playerHero, undefined, "modifier_rune_doubledamage", undefined)
-                maxLevelAbilities(playerHero)
-            }),
-            tg.immediate(() => canPlayerIssueOrders = false),
+            tg.fork([
```

Comment:
No need for fork (immediate happens instantly anyway)

Reply:
The change was to sync it a bit better with dialogue since it says: `Here are some hyper late game items for Dragon Knight as well as a few level ups. Have fun!`

Without the fork it waits for the dialogue finish before adding levels and items.

Reply:
Didn't need fork like Robinka mentions, just put the immediate before dialog to achieve the desired effect.